vivo-project/Vitro
# Issue/vivo 1947

https://jira.lyrasis.org/browse/VIVO-1947

This pull request adds safeguards to prevent both the ontology editor and the N3 editing from erroneously deleting property values in languages other than the one associated with the editing VitroRequest.

# What's new?
JenaBaseDao.updateRDFSLabel() has been updated, which in turn uses a new generalized method  JenaBaseDao.updatePlainLiteralValue().  The DAOs now use this method to update properties such as public description, example, short definition, etc.  A new ModelChangePreprocessor called LimitRemovalsToLanguage has been added to eliminate unwanted triples from the N3 editor's retractions model.  This model change preprocessor is added to each update in ProcessRdfFormController.  There are a number of other updates to various classes to use request models instead of context models and to make request-filtering DAO classes visible to the OperationController, which will try to invoke their methods via reflection.

# How should this be tested?
1. Until all labels are moved into firsttime, they are not good candidates for test.  Better to edit a class or property in the ontology editor, or open something like Site Admin and change the copyright text or change the page title on a menu page.  Save your change and then switch to a different language.  Perform the edit again, changing the value in that new language.  Switch between the two languages and observe that not only have both of your changes have been preserved, but any existing values in the remaining languages remain as well.
2. Navigate to a faculty member's profile.  Add a value for the "head of" property, entering a string in the role field.  Save the form.  Then switch languages, re-open the form and enter a new value for the role string.  Switch between languages and observe that the role strings displayed on the profile's list view track the language you have selected.
